fire/cli/niv/ilæg_revision.py

```diff
@@ -97,100 +92,351 @@ def ilæg_revision(
     opdateret = pd.DataFrame(columns=list(ARKDEF_REVISION))
 
-    # Disse navne er lange at sejle rundt med, så vi laver en kort form
-    TEKST = PunktInformationTypeAnvendelse.TEKST
-    FLAG = PunktInformationTypeAnvendelse.FLAG
-    TAL = PunktInformationTypeAnvendelse.TAL
+    # Udfyld udeladte identer
+    punkter = list(revision["Punkt"])
+    udfyldningsværdi = ""
+    for i in range(len(punkter)):
+        if punkter[i].strip() != "":
+            udfyldningsværdi = punkter[i].strip()
+            continue
+        punkter[i] = udfyldningsværdi
+    revision["Punkt"] = punkter
+
+    # Find alle punkter, der skal nyoprettes
+    oprettelse = revision.query(f"Attribut == 'Opret'")
+    for row in oprettelse.to_dict("records"):
+        if row["id"] == -1:
+            continue
+        fire.cli.print(f"Opretter nyt punkt: {row['Punkt']}")
+        if alvor:
+            opret_punkt(row["Punkt"], row["Ny værdi"], sag)
+    revision = revision.query(f"Attribut != 'Opret'")
+
+    # Find alle lokationskoordinater, der skal korrigeres
+    lokation = revision.query(f"Attribut == 'Lokation'")
+    lokation = lokation.query(f"`Ny værdi` != ''")
+    for row in lokation.to_dict("records"):
+        fire.cli.print(
+            f"IKKE IMPLEMENTERET - Korrigerer lokation for: {row['Punkt']}: {row['Ny værdi']}"
+        )
+        if alvor:
+            pass  # refaktor af opret_punkt(row["Punkt"], row["Ny værdi"], sag)
+    revision = revision.query(f"Attribut != 'Lokation'")
+
+    # Find alle koordinater, der skal oprettes
+
+    # Først skal vi bruge alle gyldige koordinatsystemnavne
+    srid_db = firedb.session.query(Srid).order_by(Srid.name).all()
+    sridnavne = [srid.name for srid in srid_db]
+
+    # Så looper vi over hele rammen og ignorerer ikke-koordinaterne
+    til_registrering = []
+    opdaterede_punkter = []
+    koordinatoprettelsestekst = str()
+    for r in revision.to_dict("records"):
+        if r["Attribut"].upper() not in sridnavne:
+            continue
+        try:
+            koord = [float(k) for k in r["Ny værdi"].split()]
+        except ValueError as ex:
+            fire.cli.print(
+                f"Slemt koordinatudtryk:\n{'    '.join(r['Ny værdi'])}\n{ex}"
+            )
+            sys.exit(1)
+
+        # Oversæt NaN til None
+        koord = [None if isnan(k) else k for k in koord]
+
+        # Tæt-på-kopi af kode fra "niv/ilæg_nye_koter.py". Her bør mediteres og overvejes
+        # hvordan denne opgave kan parametriseres på en rimeligt generel måde, så den kan
+        # udstilles i et "højniveau-API"
+        srid = firedb.hent_srid(r["Attribut"].upper())
+        registreringstidspunkt = func.current_timestamp()
+        sagsevent = Sagsevent(
+            sag=sag, id=uuid(), eventtype=EventType.KOORDINAT_BEREGNET
+        )
+
+        # Undgå forsøg på at læse punkter der endnu ikke er skrevet til databasen
+        if not alvor:
+            punkt = firedb.hent_punkt("9-09-00009")
```

Comment:
Jeg går ud fra at "9-09-00009" er et opdigtet dummy ident? Det kan ikke koliderere et faktisk ident?

Reply:
Det er @mrlarshole s testpunkt, som historisk har været brugt til tilføjelse af ting der skulle bruges for at trigge ny funktionalitet i REFGEO. I dette tilfælde skal vi bare bruge et punkt som vi ved eksisterer, uden at der er nogle særlige krav til dets øvrige natur - så jeg kunne fx også have brugt G.M.902, men det har en specifik betydning som fundamentalpunkt, hvor 9-09-00009 (alias G.I.9999) "altid" (for dem der har stredet med REFGEO) har betydet "testpunktet i det ikke eksisterende sogn".

Så man kan næppe kalde det opdigtet, men dummy er formodentlig den rette betegnelse. Jeg tilføjer en kommentar